tests/functional/object/rgw/test_rgw_virtual_hosted_ops.py

```diff
@@ -0,0 +1,157 @@
+import base64
+import logging
+
+import pytest
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.bucket_utils import (
+    verify_s3_object_integrity,
+    write_random_objects_in_pod,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_mcg_only,
+    red_squad,
+    runs_on_provider,
+    tier2,
+    rgw,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@rgw
+@runs_on_provider
+@skipif_mcg_only
+@red_squad
+class TestRgwVirtualHostedOps:
+    def prepare_cmd_and_data(
+        self, bucket_name, multipart, awscli_pod_session, test_directory_setup
+    ):
+        host_base = "rgw.data.local"
+        host_bucket = f"%(bucket)s.{host_base}"
+
+        rgw_user_creds = OCP(
+            kind="secret",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name=f"{bucket_name}",
+        ).get()
+        access_key = base64.b64decode(
+            rgw_user_creds["data"]["AWS_ACCESS_KEY_ID"]
+        ).decode("utf-8")
+        secret_key = base64.b64decode(
+            rgw_user_creds["data"]["AWS_SECRET_ACCESS_KEY"]
+        ).decode("utf-8")
+        config_cmd = (
+            f"--access_key={access_key} "
+            f"--secret_key={secret_key} "
+            f"--host={host_base} --host-bucket='{host_bucket}' --no-ssl"
+        )
+        if not multipart:
+            write_random_objects_in_pod(
+                awscli_pod_session, test_directory_setup.origin_dir, amount=1
+            )
+        else:
+            write_random_objects_in_pod(
+                awscli_pod_session, test_directory_setup.origin_dir, amount=1, bs="50M"
+            )
+            config_cmd = config_cmd + " multipart_chunk_size_mb = 5"
+        return config_cmd
+
+    @pytest.mark.parametrize(
+        argnames="multipart",
+        argvalues=[pytest.param(True), pytest.param(False)],
```

Comment:
You will need to provide the polarion decorators with a different ID for each param. See for reference on the syntax: https://github.com/red-hat-storage/ocs-ci/blob/a309162ac4b0115eb32b83689991def6a1a6a784/tests/functional/object/mcg/test_log_based_bucket_replication.py#L291-L305

Reply:
Test plan is not yet pushed to Polarion. Will update the test script once I get IDs for these test cases